chaco/segment_plot.py

```diff
@@ -351,11 +350,11 @@ def _render_icon(self, gc, x, y, width, height):
             gc.move_to(x, y)
             gc.line_to(width, height)
 
-    @on_trait_change(
+    @observe(
         'alpha, color_data:data_changed, color_mapper:updated, '
         'width_data:data_changed, width_mapper.updated, +redraw'
     )
-    def _attributes_changed(self):
+    def _attributes_changed(self, event):
```

Comment:
same comment as above about renaming this to `_invalidate_and_redraw` if possible